cheetah/bmadx_utils.py

```diff
@@ -0,0 +1,148 @@
+import torch
+
+
+def cheetah_to_bmad_coords(
+    cheetah_coords: torch.Tensor, ref_energy: torch.Tensor, mc2: torch.Tensor
+) -> torch.Tensor:
+    """Transforms Cheetah coordinates to Bmad coordinates.
+    :param cheetah_coords: 7-dimensional particle vectors in Cheetah coordinates.
+    :param ref_energy: reference energy in eV.
+    """
+    # initialize Bmad coordinates:
+    bmad_coords = cheetah_coords[..., :6].clone()
+
+    # Cheetah longitudinal coords:
+    tau = cheetah_coords[..., 4]
+    delta = cheetah_coords[..., 5]
+
+    # compute p0c and bmad z, pz
+    p0c = torch.sqrt(ref_energy**2 - mc2**2)
+    energy = ref_energy + delta * p0c
+    p = torch.sqrt(energy**2 - mc2**2)
+    beta = p / energy
+    z = -beta * tau
+    pz = (p - p0c) / p0c
+
+    # Bmad coords:
+    bmad_coords[..., 4] = z
+    bmad_coords[..., 5] = pz
+
+    return bmad_coords, p0c
+
+
+def bmad_to_cheetah_coords(
+    bmad_coords: torch.Tensor, p0c: torch.Tensor, mc2: torch.Tensor
+) -> torch.Tensor:
```

Comment:
Would it make sense to just integrate these two methods also in `ParticleBeam` like the other transformations?
I think `p0c` can also be provided as a property.

Reply:
I was thinking to include the coordinate transforms in here instead: https://github.com/desy-ml/cheetah/blob/master/cheetah/converters/bmad.py

This was it could also be used to transform beams for both Bmad and Bmad-X. 

Any thoughts?